static/src/javascripts/bootstraps/enhanced/youtube.js

```diff
@@ -0,0 +1,92 @@
+define([
+    'fastdom',
+    'common/modules/video/youtube-player',
+    'common/modules/video/youtube-tracking',
+    'common/utils/$'
+], function (
+    fastdom,
+    youtubePlayer,
+    tracking,
+    $
+) {
+
+    var STATES = {
+        'ENDED': _onPlayerEnded,
+        'PLAYING': _onPlayerPlaying,
+        'PAUSED': _onPlayerPaused
+    };
+
+    var progressTracker = {};
+    var players = {};
+
+    function init() {
+
+        fastdom.read(function () {
+                        $('.atom--media--youtube').each(function (el) {
+                            var atomId = el.getAttribute('data-media-atom-id');
+                            var videoId = el.firstElementChild.id;
+                            tracking.init(atomId);
+                            players[videoId] = youtubePlayer.init(el,
+                                {
+                                    onPlayerStateChange: function (event) {
+                                        Object.keys(STATES).forEach(checkState.bind(null, videoId, atomId, event.data, event.target));
+                                    },
+                                    onPlayerReady: function (event) {
+                                        var player = event.target;
+                                        // Record the duration for percentage calculation.
+                                        player.duration = player.getDuration();
```

Comment:
Is this useful? We only access this `duration` property once from within the `recordPlayerProgress` function below, but we could just as easily call `getDuration()` there instead